src/analysis/box_integral/partition/additive.lean

```diff
@@ -0,0 +1,608 @@
+such that `π.compl.Union = I \ π.Union`. -/
+def compl (π : prepartition I) : prepartition I := π.exists_Union_eq_diff.some
+
+@[simp] lemma Union_compl (π : prepartition I) : π.compl.Union = I \ π.Union :=
+π.exists_Union_eq_diff.some_spec
+
+/-- Since the definition of `box_integral.prepartition.compl` uses `Exists.some`,
+the result depends only on `π.Union`. -/
+lemma compl_congr {π₁ π₂ : prepartition I} (h : π₁.Union = π₂.Union) :
+  π₁.compl = π₂.compl :=
+by { dunfold compl, congr' 1, rw h }
+
+lemma is_partition.compl_eq_bot {π : prepartition I} (h : is_partition π) : π.compl = ⊥ :=
+by rw [← Union_eq_empty, Union_compl, h.Union_eq, diff_self]
+
+@[simp] lemma compl_top : (⊤ : prepartition I).compl = ⊥ := (is_partition_top I).compl_eq_bot
+
+end fintype
+
+end prepartition
+
+/-- A function on `box ι` is called box additive if for every box `I` and a hyperplane
+`{y | y i = x}` we have `f (I ∩ {y | y i ≤ x}) + f (I ∩ {y | x < y i}) = f I`. We formualte this
+property in terms of `box_integral.box.split_lower` and `box_integral.box.split_upper`.
+
+This property implies that for any partition `π` of a box `I` into smaller boxes we have
+`f I = ∑ J in π.boxes, f J`. -/
+structure box_additive_map (ι M : Type*) [add_comm_monoid M] (I : with_top (box ι)) :=
+(to_fun : box ι → M)
+(sum_partition_boxes' : ∀ J : box ι, ↑J ≤ I → ∀ π : prepartition J, π.is_partition →
+  ∑ J' in π.boxes, to_fun J' = to_fun J)
+
+notation ι ` →ᵇᵃ `:25 M := box_additive_map ι M ⊤
+notation ι ` →ᵇᵃ[`:25 I `] ` M := box_additive_map ι M I
+
+namespace box_additive_map
+
+open box prepartition finset
+
+variables {N : Type*} [add_comm_monoid M] [add_comm_monoid N] {I₀ : with_top (box ι)}
+  {I J : box ι} {i : ι}
+
+instance : has_coe_to_fun (ι →ᵇᵃ[I₀] M) := ⟨_, to_fun⟩
+
+initialize_simps_projections box_integral.box_additive_map (to_fun → apply)
+
+@[simp] lemma to_fun_eq_coe (f : ι →ᵇᵃ[I₀] M) : f.to_fun = f := rfl
+
+@[simp] lemma coe_mk (f h) : ⇑(mk f h : ι →ᵇᵃ[I₀] M) = f := rfl
+
+lemma coe_injective : injective (λ (f : ι →ᵇᵃ[I₀] M) x, f x) :=
+by { rintro ⟨f, hf⟩ ⟨g, hg⟩ (rfl : f = g), refl }
+
+@[simp] lemma coe_inj {f g : ι →ᵇᵃ[I₀] M} : (f : box ι → M) = g ↔ f = g :=
+coe_injective.eq_iff
+
+lemma sum_partition_boxes (f : ι →ᵇᵃ[I₀] M) (hI : ↑I ≤ I₀) {π : prepartition I}
+  (h : π.is_partition) :
+  ∑ J in π.boxes, f J = f I :=
+f.sum_partition_boxes' I hI π h
+
+@[simps { fully_applied := ff }] instance : has_zero (ι →ᵇᵃ[I₀] M) :=
+⟨⟨0, λ I hI π hπ, sum_const_zero⟩⟩
+
+instance : inhabited (ι →ᵇᵃ[I₀] M) := ⟨0⟩
+
+instance : has_add (ι →ᵇᵃ[I₀] M) :=
+⟨λ f g, ⟨f + g, λ I hI π hπ,
+  by simp only [pi.add_apply, sum_add_distrib, sum_partition_boxes _ hI hπ]⟩⟩
+
+instance : add_comm_monoid (ι →ᵇᵃ[I₀] M) :=
+function.injective.add_comm_monoid _ coe_injective rfl (λ _ _, rfl)
+
+@[simp] lemma map_split_add (f : ι →ᵇᵃ[I₀] M) (hI : ↑I ≤ I₀) {x : ℝ} (hl : I.lower i < x)
+  (hu : x < I.upper i) :
```

Comment:
In all your other lemmas above the assumptions were as `x \in Ioo I.lower I.upper`. Is there a reason you changed here?